scripts/deploy.js

```diff
@@ -0,0 +1,54 @@
+var NodeGit = require(`nodegit`);
+var shell = require(`shelljs`);
+var pathToRepo = require(`path`).resolve(`.`);
+
+var getStatus = (repo) => {
+  return repo.getStatus();
+};
+
+var runDeploy = () => {
+  shell.echo(`Running deployment now...`);
+
+  shell.exec(`git checkout master`);
+  shell.exec(`git pull origin master`);
+
+  shell.rm(`-rf`,`node_modules`);
+
+  shell.exec(`npm i`);
+  shell.exec(`npm run build`);
+
+  shell.exec(`git branch -D gh-pages`);
+  shell.exec(`git checkout --orphan gh-pages`);
+
+  shell.rm(`.gitignore`);
+
+  shell.echo(`/*\n`).toEnd(`.gitignore`);
+  shell.echo(`!css\n`).toEnd(`.gitignore`);
+  shell.echo(`!images\n`).toEnd(`.gitignore`);
+  shell.echo(`!index.html\n`).toEnd(`.gitignore`);
+  shell.echo(`!last-built.txt\n`).toEnd(`.gitignore`);
+
+  shell.echo(new Date).to(`last-built.txt`);
+
+  shell.mv(`dest/*`, `./`);
+
+  shell.exec(`git reset`);
+  shell.exec(`git add .`);
+  shell.exec(`git commit -m 'deploy.js-ified'`);
```

Comment:
Since it is a website and not an app or library I don't see much value in version numbers. However, I could add the latest commit hash to the `last-built.txt` which basically functions as a healthcheck.